src/utils/network-monitor.ts

```diff
@@ -901,26 +841,25 @@ export class NetworkMonitor {
   }
 
   restartProvider = async (network: string): Promise<void> => {
-    const rpcEndpoint = (this.configFile.networks[network as keyof ConfigNetworks] as ConfigNetwork).providerUrl
+    const networkConfig = this.configFile.networks[network as keyof ConfigNetworks] as ConfigNetwork
+    const rpcEndpoint = networkConfig.providerUrl
     const protocol = new URL(rpcEndpoint).protocol
+
     switch (protocol) {
       case 'http:':
-        this.providers[network] = new JsonRpcProvider(rpcEndpoint)
-
-        break
       case 'https:':
         this.providers[network] = new JsonRpcProvider(rpcEndpoint)
 
         break
       case 'wss:':
-        this.ws[network].close(1012, 'Block Job Handler has been inactive longer than threshold time.')
-        //  this.failoverWebSocketProvider.bind(this)(network, rpcEndpoint, false)
+        this.ws[network]?.close(1012, 'Block Job Handler has been inactive longer than threshold time.')
```

Comment:
@alexanderattar ah nvm, i see the websocket connector is still here.

Reply:
Let's keep this for now 👍